src/transformers/models/trillsson_efficientnet/feature_extraction_trillsson_efficientnet.py

```diff
@@ -0,0 +1,513 @@
+        num_mel_bins (`int`, *optional*, defaults to 80):
+            Number of Mel bins.
+        log_floor (`float`, *optional*, defaults to 1e-12):
+            Floor on the log-mel spectrogram.
+        log_additive_offset (`float`, *optional*, defaults to 0.001):
+            Offset on the log-mel spectrogram.
+        window_length_secs (`float`, *optional*, defaults to 0.025):
+            Window length in seconds.
+        hop_length_secs (`float`, *optional*, defaults to 0.010):
+            Hop length in seconds.
+        f_max (`float`, *optional*, defaults to 7500.0):
+            Maximum frequency for the Mel filterbank.
+        f_min (`float`, *optional*, defaults to 125.0):
+            Minimum frequency for the Mel filterbank.
+        mel_high_frequency_q (`float`, *optional*, defaults to 1127.0):
+            Constant used in the mel scale.
+        mel_break_frequency_hertz (`float`, *optional*, defaults to 700.0):
+            Constant used in the mel scale.
+        fft_length (`int`, *optional*):
+            FFT length. If `None`, it will be computed from the window length. Defaults to `None`.
+    """
+
+    model_input_names = ["input_values"]
+
+    def __init__(
+        self,
+        feature_size=1,
+        sampling_rate=16000,
+        padding_value=0.0,
+        do_normalize=True,
+        return_attention_mask=False,
+        num_mel_bins=80,
+        log_floor=1e-12,
+        log_additive_offset=0.001,
+        window_length_secs=0.025,
+        hop_length_secs=0.010,
+        f_max=7500.0,
+        f_min=125.0,
+        mel_high_frequency_q=1127.0,
+        mel_break_frequency_hertz=700.0,
+        fft_length=None,
+        **kwargs
+    ):
+        super().__init__(feature_size=feature_size, sampling_rate=sampling_rate, padding_value=padding_value, **kwargs)
+        self.return_attention_mask = return_attention_mask
+        self.do_normalize = do_normalize
+        self.num_mel_bins = num_mel_bins
+        self.log_floor = log_floor
+        self.log_additive_offset = log_additive_offset
+        self.window_length_secs = window_length_secs
+        self.hop_length_secs = hop_length_secs
+        self.f_max = f_max
+        self.f_min = f_min
+        self.mel_high_frequency_q = mel_high_frequency_q
+        self.mel_break_frequency_hertz = mel_break_frequency_hertz
+        self.fft_length = fft_length
+
+    @classmethod
+    def pad_symmetric(cls, raw_speech: np.ndarray, padding: int):
+        """Symmetric pad a 2D Tensor"""
+        if raw_speech.ndim != 2:
+            raise ValueError(f"Raw speech must be rank 2 got {raw_speech.shape}")
+        raw_speech_len = raw_speech.shape[1]
+        if padding > raw_speech_len:
+            pad_raw_speech = cls.repeat_n_times_with_extra(raw_speech, padding, raw_speech_len)
+        else:
+            pad_raw_speech = np.pad(raw_speech, [(0, 0), (0, padding)], mode="symmetric")
+        return pad_raw_speech
```

Comment:
The super class already has a `pad` method which supports symetric padding. Is there a particular reason for the re-implementation?

Reply:
I have checked the super Class `pad` method, and currently, I didn't see any support for padding with `mode=symmetric `.https://github.com/huggingface/transformers/blob/59d5edef34ae0fa56065a2e863736d4f133c558b/src/transformers/feature_extraction_sequence_utils.py#L226-L293. So, I followed the padding strategy from the original repo https://github.com/google-research/google-research/blob/e5c2af8f5be69e5e6cbe523a16f48807d92560b5/non_semantic_speech_benchmark/export_model/tf_pad.py#L34-L41